trakanom/Portfolio-Site
# Analytics, Consent & Styling Updates

**Description**:
This PR introduces several key updates to improve user experience, ensure GDPR compliance, and enhance the overall look and feel of the application.

**Key Changes**:

1. **Analytics Integration**:
   - Integrated Google Tag Manager for better tracking and analytics.
   - Added logic to ensure user consent before initializing analytics.

2. **Consent Banner**:
   - Introduced a new `ConsentBanner` component to inform users about cookie usage and analytics tracking.
   - Used local storage to remember user's consent choice and ensure the banner is not shown repeatedly to the same user.
   - Refactored consent logic into a custom hook (`useAnalyticsConsent`) for better modularity and reusability.

3. **Styling Updates**:
   - Minor styling enhancements to improve the visual appeal of the consent banner.
   - Adjusted positioning to ensure the banner is always visible at the bottom of the user's screen.

**Files Changed**:
- `App.js`
- `ConsentBanner.js`
- `UnderConstructionSplash.js`
- `useAnalyticsConsent.js`
- `App.css`

**How to Test**:
1. Pull the latest changes from the `feature/analytics-consent-styling` branch.
2. Run the application locally.
3. Ensure the consent banner appears at the bottom of the screen.
4. Test the "Accept" and "Decline" actions on the consent banner.
5. Verify that user's choice is remembered across sessions (using local storage).
6. Navigate through different pages and ensure styling consistency.
